lib/internal/test_runner/runner.js

```diff
@@ -765,9 +765,15 @@ function run(options = kEmptyObject) {
             SafePromiseAllReturnVoid([root.harness.bootstrapPromise, promise]) :
             promise;
 
-          const userImports = getOptionValue('--import');
-          for (let i = 0; i < userImports.length; i++) {
-            await cascadedLoader.import(userImports[i], parentURL, kEmptyObject);
+          // We need to setup the user modules in the test runner if we are running with
+          // --test-isolation=none and --test in order to avoid loading the user modules
+          // before the creation of the root test.
+          const userModules = [
+            ...getOptionValue('--require'),
+            ...getOptionValue('--import'),
+          ];
```

Comment:
I think the approach here is logical in terms of steps. But 2 caveats:

* `CascadedLoader` (aka `ESMLoader` aka `ModuleLoader`) does not handle CJS modules exactly the same as `CJSLoader`; if we're in ESM mode, this is "correct" (ESMLoader should handle it); but if not ESM mode, for good or ill, it could lead to unexpected behaviour.
* Does this prevent the `--require`d module(s) executing prior to this? If not, they could be stateful or something else crazy which would get lost when the clone gets created here.

Reply:
Thanks for your review, @JakobJingleheimer 🚀 

Regarding nr1: ok! I suspected that. I'll take a look to understand if we can just use the default method we're using for `--require` in the "default" entry point.

nr2: Yes, it prevents the default handling of `--require` from happening, it's now being executed only at this point!

Reply:
W00t!

Reply:
I've updated the logic to use `loadPreloadModules` and applied the other suggestions.  
Before marking this PR as "ready to be reviewed", I'd like to ask you to take another look at these changes 😁